### ISSUE TITLE
doxygen and pytorch documentation

### ISSUE THREAD

**Issue by stephanecharette** [pytorch/pytorch#53157]
## 📚 Documentation

PyTorch needs to have better documentation.  The web site hosts document pages which are essentially blank and offers zero information:  https://pytorch.org/cppdocs/api/classat_1_1_tensor.html

I attempted to generate my own Doxygen docs, but quickly realized that output of my own doxygen pages looked essentially the same as what is available on the web site.

This makes it extremely difficult for developers new to pytorch to use the API, as it appears there is no documentation available.

**Comment by gchanan** [comment]
just to check -- you are talking specifically about the C++ documentation?

**Comment by stephanecharette** [comment]
Yes, the documentation at the link I provided above.  I'm thinking that maybe that is the wrong link?

**Comment by mruberry** [comment]
The frontpage for the Python API is https://pytorch.org/docs/master/, and the frontpage for the C++ API is https://pytorch.org/cppdocs/. The C++ API page links to some examples that might help with getting started, although the documentation could definitely be improved.

Just to double-check: you are interested in using PyTorch's C++ API, right?

**Comment by glaringlee** [comment]
@stephanecharette 
The pytorch c++ doc is actually generated from doxygen and converted to html, so you get exactly the same document when doing your own doxygen. The c++ doc needs to be improved for sure, in the mean time, you can take a look at our test files which might be more strightforward.
https://github.com/pytorch/pytorch/tree/master/test/cpp/api
